### ISSUE TITLE
meta.categories contains null value if media:category is empty

### ISSUE THREAD

**Issue by EggiJan** [danmactough/node-feedparser#136]
I just recognised, that while parsing a feed with an empty media:category tag, the feedparser returns an array that contains a null value as meta.categories.

For example like in this feed: http://feeds.feedburner.com/outofschool

```
<itunes:category text="Education">
    <itunes:category text="Education Technology" />
</itunes:category>
<itunes:category text="">
</itunes:category>
<itunes:category text="">
</itunes:category>
<media:category scheme="http://www.itunes.com/dtds/podcast-1.0.dtd">Education/Education Technology</media:category>
<media:category scheme="http://www.itunes.com/dtds/podcast-1.0.dtd"></media:category>
<media:category scheme="http://www.itunes.com/dtds/podcast-1.0.dtd"></media:category>
```

```
categories: [ 'Education', 'Education/Education Technology', null ]
```

Is this behaviour intended?

**Comment by danmactough** [comment]
Not intended. I would expect the correct value to be `[ 'Education', 'Education/Education Technology' ]`.

Agreed?

**Comment by EggiJan** [comment]
Yes

**Comment by danmactough** [comment]
Published at v1.0.1 Thanks @EggiJan for reporting the issue.
